lib/mergeDeepWithKey/index.js

```diff
@@ -0,0 +1,10 @@
+import _curry3 from '../_internal/_curry3'
+import _merge from '../_internal/_merge'
+
+export default _curry3(function mergeDeepWithKey(func, left, right) {
+  var cb = function(l, r, key) {
+    if (typeof l === 'object' && typeof r === 'object' && !Array.isArray(l)) return mergeDeepWithKey(func, l, r)
```

Comment:
sorry, how could you imagine `l` to be `object` and `array` at the same time?
I mean `typeof l === 'object'` and `!Array.isArray(l)`

Reply:
You forgot to use `keys`

Reply:
`typeof` of array and object both will be 'object'
<img width="124" alt="new_tab" src="https://user-images.githubusercontent.com/25172855/39052662-6ea55aa8-44b5-11e8-8d57-0564c09b20d2.png">

Reply:
> You forgot to use `keys`

Where?

Reply:
@BabkinAleksandr oops, agree, sorry